Tasks/Gradle/CodeAnalysis/SonarQube/common.ts

```diff
@@ -21,145 +23,42 @@ export function isSonarQubeAnalysisEnabled(): boolean {
+ */
+export function applySonarQubeParameters(toolRunner: ToolRunner): ToolRunner {
+    return SonarQubeParameterHelper.applySonarQubeParameters(toolRunner);
 }
 
 // Upload a build summary with links to available SonarQube dashboards for further analysis details.
-export function uploadSonarQubeBuildSummaryIfEnabled(sqBuildFolder:string):Q.Promise<void> {
-    return createSonarQubeBuildSummary(sqBuildFolder)
-        .then((buildSummaryContents:string) => {
-            var buildSummaryFilePath = saveSonarQubeBuildSummary(buildSummaryContents);
-            tl.debug('Uploading build summary from ' + buildSummaryFilePath);
-
-            tl.command('task.addattachment', {
-                'type': 'Distributedtask.Core.Summary',
-                'name': tl.loc('sqAnalysis_BuildSummaryTitle')
-            }, buildSummaryFilePath);
-        });
-}
-
-// Gets a SonarQube authentication parameter from the specified connection endpoint.
-// The endpoint stores the auth details as JSON. Unfortunately the structure of the JSON has changed through time, namely the keys were sometimes upper-case.
-// To work around this, we can perform case insensitive checks in the property dictionary of the object. Note that the PowerShell implementation does not suffer from this problem.
-// See https://github.com/Microsoft/vso-agent/blob/bbabbcab3f96ef0cfdbae5ef8237f9832bef5e9a/src/agent/plugins/release/artifact/jenkinsArtifact.ts for a similar implementation
-function getSonarQubeAuthParameter(endpoint: string, paramName: string) {
-
-    var paramValue = null;
-    var auth = tl.getEndpointAuthorization(endpoint, false);
-
-    if (auth.scheme != "UsernamePassword") {
-        throw new Error("The authorization scheme " + auth.scheme + " is not supported for a SonarQube endpoint. Please use a username and a password.");
-    }
-
-    var parameters = Object.getOwnPropertyNames(auth['parameters']);
-
-    var keyName;
-    parameters.some(function (key) {
-
-        if (key.toLowerCase() === paramName.toLowerCase()) {
-            keyName = key;
-
-            return true;
-        }
-    });
-
-    if (keyName) {
-        paramValue = auth['parameters'][keyName];
-    }
-
-    return paramValue;
+/**
+ * Act on all user-enabled SonarQube integration options: build break, build summary, etc.
+ * @param sqBuildFolder
+ * @returns {Promise<void>} Promise resolved when all SQ integration actions are complete.
+ */
+export function processSonarQubeIntegration(sqBuildFolder:string):Q.Promise<void> {
+    var sqRunSettings:SonarQubeRunSettings = getSonarQubeRunSettings(sqBuildFolder);
+    var sqMetrics:SonarQubeMetrics = getSonarQubeMetrics(sqRunSettings);
+
+    // Wait for all promises to complete before proceeding (even if one or more promises reject).
+    return Q.allSettled([
+        processSonarQubeBuildSummary(sqRunSettings, sqMetrics),
+        processSonarQubeBuildBreaker(sqRunSettings, sqMetrics),
+    ])
+        .then(() => {
+        })
```

Comment:
Previously, the only integration task was the upload the build summary. Now that we have more, the public entry point will be this function _processSonarQubeIntegration()_, which will act on all enabled integration actions.